pkg/controllers/follower/controller.go

```diff
@@ -134,160 +126,201 @@ func (c *Controller) IsControllerReady() bool {
-			func(qualifiedName common.QualifiedName) worker.Result {
-				return reconcile(handles, qualifiedName)
-			},
-			worker.RateLimiterOptions{},
-			workerCount,
-			c.metrics,
-			delayingdeliver.NewMetricTags("follower-controller-worker", handles.name),
-		)
-		handles.informer.Informer().AddEventHandlerWithResyncPeriod(
-			util.NewTriggerOnAllChanges(handles.worker.EnqueueObject),
-			util.NoResyncPeriod,
-		)
-
-		return handles
-	}
-
-	ftcs, err := c.fedClient.CoreV1alpha1().
-		FederatedTypeConfigs().
-		List(context.TODO(), metav1.ListOptions{})
-	if err != nil {
-		return nil, err
 	}
 
-	// Find the supported leader and follower types and create their handles
-	for i := range ftcs.Items {
-		ftc := &ftcs.Items[i]
-		targetType := ftc.Spec.TargetType
-		federatedType := ftc.Spec.FederatedType
+	c.informerManager.AddFTCUpdateHandler(func(_, latest *fedcorev1a1.FederatedTypeConfig) {
+		targetType := latest.Spec.SourceType
 		targetGK := schema.GroupKind{Group: targetType.Group, Kind: targetType.Kind}
-		federatedGK := schema.GroupKind{Group: federatedType.Group, Kind: federatedType.Kind}
 
 		if _, exists := leaderPodTemplatePaths[targetGK]; exists {
-			handles := getHandles(ftc, "leader", c.reconcileLeader)
-			c.sourceToFederatedGKMap[targetGK] = federatedGK
-			c.leaderTypeHandles[federatedGK] = handles
-			c.logger.V(2).Info(fmt.Sprintf("Found supported leader FederatedTypeConfig %s", ftc.Name))
+			handles := getHandles(latest, "leader")
+			c.leaderTypeHandles[targetGK] = handles
+			c.logger.V(2).Info(fmt.Sprintf("Found supported leader FederatedTypeConfig %s", latest.Name))
 		} else if supportedFollowerTypes.Has(targetGK) {
-			handles := getHandles(ftc, "follower", c.reconcileFollower)
-			c.sourceToFederatedGKMap[targetGK] = federatedGK
-			c.followerTypeHandles[federatedGK] = handles
-			c.logger.V(2).Info(fmt.Sprintf("Found supported follower FederatedTypeConfig %s", ftc.Name))
+			handles := getHandles(latest, "follower")
+			c.followerTypeHandles[targetGK] = handles
+			c.logger.V(2).Info(fmt.Sprintf("Found supported follower FederatedTypeConfig %s", latest.Name))
 		}
-	}
+	})
 
 	return c, nil
 }
 
-func (c *Controller) Run(stopChan <-chan struct{}) {
-	c.logger.Info("Starting controller")
-	defer c.logger.Info("Stopping controller")
+func (c *Controller) Run(ctx context.Context) {
+	ctx, logger := logging.InjectLogger(ctx, c.logger)
+
+	logger.Info("Starting controller")
+	defer logger.Info("Stopping controller")
+
+	if !cache.WaitForNamedCacheSync(ControllerName, ctx.Done(), c.HasSynced) {
+		logger.Error(nil, "Timed out waiting for caches to sync")
+		return
+	}
+	logger.Info("Caches are synced")
+	c.worker.Run(ctx)
+	<-ctx.Done()
+}
+
+func (c *Controller) HasSynced() bool {
+	return c.informerManager.HasSynced() &&
+		c.fedObjectInformer.Informer().HasSynced() &&
+		c.clusterFedObjectInformer.Informer().HasSynced()
+}
+
+func (c *Controller) NewTriggerWithSupportedTypeFilter() *cache.ResourceEventHandlerFuncs {
```

Comment:
Can this be simplified to `eventhandlers.NewTriggerOnAllChangs(c.enqueueSupportedType)`